infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/SnapshotSyncUtils.java

```diff
@@ -93,4 +99,80 @@ public static void enforceSnapshotSync(LogReplicationSession session, CorfuStore
             throw new UnrecoverableCorfuInterruptedError(e);
         }
     }
+
+    public static void removeUpgradeSnapshotSyncEvent(LogReplicationEvent event, CorfuStore corfuStore) {
+        Table<LogReplicationMetadata.ReplicationEventInfoKey, LogReplicationMetadata.ReplicationEvent, Message> replicationEventTable =
+                tryOpenTable(corfuStore, CORFU_SYSTEM_NAMESPACE, REPLICATION_EVENT_TABLE_NAME,
+                        ReplicationEventInfoKey.class, ReplicationEvent.class, null);
+
+        List<CorfuStoreEntry<ReplicationEventInfoKey, ReplicationEvent, Message>> events;
+        try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
+            events = txn.executeQuery(REPLICATION_EVENT_TABLE_NAME, p -> true);
+            txn.commit();
+        } catch (Exception e) {
+            log.error("Failed to get the replication events", e);
+            return;
+        }
+
+        try (TxnContext txnContext = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            boolean entryFound = false;
+            for (CorfuStoreEntry<ReplicationEventInfoKey, ReplicationEvent, Message> entry : events) {
+                if (entry.getPayload().getEventId().equals(String.valueOf(event.getMetadata().getRequestId()))) {
+                    txnContext.delete(replicationEventTable, entry.getKey());
+                    entryFound = true;
+                    break;
+                }
+            }
+
+            if (entryFound) {
+                log.info("Upgrade forced sync event removed, request id: {}", event.getMetadata().getRequestId());
+            } else {
+                log.info("No force sync event found with matching request id: {}", event.getMetadata().getRequestId());
+            }
+            txnContext.commit();
+        } catch (TransactionAbortedException e) {
+            log.error("TX Abort while trying to remove force sync event.", e);
+        }
+    }
+
+    // TODO (Shreay): Verify all possible paths for force sync re-triggers
+    public static void modifyUpgradeSnapshotSyncEvent(UUID oldEventId, UUID newEventID, CorfuStore corfuStore) {
```

Comment:
what is the purpose of this method?

Reply:
The purpose of this method is to modify the uuid of the rolling upgrade force sync event in the event table so we can find it later incase it is re-triggered and assigned a new uuid.

Reply:
ok.  In that case, let us discuss if it is better to use the session to identify the event instead of the event id.